sdk/formrecognizer/Azure.AI.FormRecognizer/tests/samples/Receipt.cs

```diff
@@ -0,0 +1,216 @@
+
+        /// <summary>
+        /// The transaction date.
+        /// </summary>
+        public FormField<DateTime> TransactionDate { get; }
+
+        /// <summary>
+        /// The transaction time.
+        /// </summary>
+        public FormField<TimeSpan> TransactionTime { get; }
+
+        /// <summary>
+        /// Converts an item <see cref="FormField"/> into a list of <see cref="ReceiptItem"/>, making the name,
+        /// quantity, price and total price of each item strongly-typed. Returns an empty list if the "Items"
+        /// field does not exist or if conversion is not possible.
+        /// </summary>
+        /// <param name="fields">The dictionary of fields in the recognized receipt. Returned by the service.</param>
+        /// <returns>A list of <see cref="ReceiptItem"/> with strongly-typed properties. Returns an empty list if the "Items" field does not exist or if conversion is not possible.</returns>
+        private IReadOnlyList<ReceiptItem> ConvertReceiptItems(IReadOnlyDictionary<string, FormField> fields)
+        {
+            List<ReceiptItem> items = new List<ReceiptItem>();
+
+            if (fields.TryGetValue("Items", out FormField itemsField)
+                && itemsField.Value.Type == FieldValueType.List)
+            {
+                foreach (FormField itemField in itemsField.Value.AsList())
+                {
+                    if (itemField.Value.Type == FieldValueType.Dictionary)
+                    {
+                        IReadOnlyDictionary<string, FormField> itemPropertyFields = itemField.Value.AsDictionary();
+
+                        FormField<string> name = ConvertStringField("Name", itemPropertyFields);
+                        FormField<float> quantity = ConvertFloatField("Quantity", itemPropertyFields);
+                        FormField<float> price = ConvertFloatField("Price", itemPropertyFields);
+                        FormField<float> totalPrice = ConvertFloatField("TotalPrice", itemPropertyFields);
+
+                        items.Add(new ReceiptItem(name, quantity, price, totalPrice));
+                    }
+                }
+            }
+
+            return items;
+        }
+
+        /// <summary>
+        /// Converts a weakly-typed <see cref="FormField"/> into a strongly-typed <see cref="FormField{T}"/> of type
+        /// <c>string</c>. Returns <c>null</c> if field does not exist or if conversion is not possible.
+        /// </summary>
+        /// <param name="fieldName">The key to retrieve the field to be converted from the <paramref name="fields"/> dictionary.</param>
+        /// <param name="fields">The dictionary of fields in the recognized receipt. Returned by the service.</param>
+        /// <returns>A strongly-typed <see cref="FormField{T}"/>. Returns <c>null</c> if field does not exist or if conversion is not possible.</returns>
+        private static FormField<string> ConvertStringField(string fieldName, IReadOnlyDictionary<string, FormField> fields)
```

Comment:
I haven't included the `ConvertInt64Field` because they're not used in receipts at all. Wondering if it would be worth to add it only for users to copy-paste them into the model they create for their custom forms.

If we decide not to include it, should we mention it in the sample?

Reply:
I like the idea of including it so users get everything in copy-paste.